app/controllers/providerPartnership.js

```diff
@@ -554,28 +554,53 @@ exports.newProviderPartnershipCheck_post = async (req, res) => {
 /// ------------------------------------------------------------------------ ///
 
 exports.deleteProviderPartnership_get = async (req, res) => {
-  // get the provider and partnership IDs from the request
   const { providerId, partnershipId } = req.params
+
   const provider = await Provider.findByPk(providerId)
-  const partnership = await ProviderPartnership.findByPk(partnershipId, {
+
+  const partnership = await ProviderAccreditationPartnership.findByPk(partnershipId, {
     include: [
       {
-        model: Provider,
-        as: 'trainingProvider'
+        model: ProviderAccreditation,
+        as: 'providerAccreditation',
+        include: [
+          {
+            model: Provider,
+            as: 'provider'
+          }
+        ]
       },
       {
         model: Provider,
-        as: 'accreditedProvider'
+        as: 'partner'
       }
     ]
   })
 
-  const isAccredited = await isAccreditedProvider({ providerId })
+  if (!provider || !partnership) {
+    return res.status(404).render('errors/404')
+  }
+
+  // Determine relationship direction
```

Comment:
Getting messhy to follow

Maybe it's a ProviderAccreditationPartnership model should be introduced to capture the determination of this.

Reply:
Yeah, prototype code is generally messy.

There is likely a task to refactor a significant portion of the code to improve it.